qiskit_experiments/framework/restless_mixin.py

```diff
@@ -91,16 +96,21 @@ def enable_restless(
         # The excited state promotion readout analysis option is set to
         # False because it is not compatible with a restless experiment.
         if self._t1_check(rep_delay):
+            meas_level = self._default_run_options().get("meas_level", MeasLevel.CLASSIFIED)
+            meas_return = self._default_run_options().get("meas_return", MeasReturnType.SINGLE)
```

Comment:
I think this doesn't work because user can `.run` experiment with run option. Likely it enforces user to use default run option since it implicitly overrides options that user may set with `set_run_options` after one created experiment instance.

Perhaps you can implement following logic
1. Override `_default_run_options` with `restless=False`.
2. When `enable_restless` is called it just sets the flag `restless=True` in run option.
3. Override `_finalize` method to check the flag. If flag is `True` then call this routine with fresh run option from `self.run_options`.

This logic should be implemented so that
```python
exp = FineXAmplitude(qubit=0)
exp. enable_restless()
exp.run(backend, meas_level=2)
```
this case works as expected.

Reply:
Yes, thanks, this will not work.
I opened an issue describing the problem: https://github.com/Qiskit/qiskit-experiments/issues/795. Should we maybe solve this problem in an additional PR?